File: HPC/Organized/archive/number_guessing/number_guessing.py
#!/usr/bin/python3
import random
# generate a number from 1 to 100
gen = random.randint( 1, 100 )
inp = None
c = 0

# while input != generated
while inp != gen:
    inp = int(input("Tell me your guess: "))
    c += 1
    if inp > gen:
        print("Need smaller number!")
    elif inp < gen:
        print("Need bigger number!")

print(f'You guessed the number! You needed {c} guesses.')

# ask input from user
# if equal, print message and exit
# otherwise goto ask input

# end while

File: HPC/Organized/archive/number_guessing/number_guessing.c
#include <stdio.h>
#include <stdlib.h>
#include <time.h>

int main()
{
    srand( time( NULL ) );
    int gen = rand() % 100 + 1;
    int inp = -1;
    int c = 0;

    while( inp != gen )
    {
        printf("Tell me your guess: ");
        scanf("%d", &inp);
        c++;
        if( inp > gen )
            printf("Need smaller number!\n");
        else if( inp < gen )
            printf("Need bigger number!\n");
    }
    printf("You guessed the number! You needed %d guesses.\n", c);
    return 0;
}
